File: PythonCode/lintcode/easy/SqrtSolution.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

# @Time    : 2018/2/10 16:22
# @Author  : bearyang
# @Site    : 
# @File    : SqrtSolution.py
# @Software: PyCharm
# 141. x的平方根
# http://www.lintcode.com/zh-cn/problem/sqrtx/


class Solution:
    """
    @param: x: An integer
    @return: The sqrt of x
    """
    def sqrt(self, x):
        if x == 0:
            return 0
        begin = 1
        end = x
        middle = (begin + end) / 2
        while begin <= end:
            if x / int(middle) == int(middle):
                return int(middle)
            elif x / int(middle) > int(middle):
                begin = int(middle) + 1
            else:
                end = int(middle) - 1
            middle = (begin + end) / 2
        return int(middle)


test = Solution()
print(test.sqrt(3))
print(test.sqrt(8))
print(test.sqrt(9))
print(test.sqrt(11))

File: JsCode/lintcode/easy/SqrtSolution.js
/**
 * 141. x的平方根
 * http://www.lintcode.com/zh-cn/problem/sqrtx/
 * @param x: An integer
 * @return: The sqrt of x
 */
const sqrt = function (x) {

    if(x === 0){
        return 0;
    }
    var begin = 1;
    var end = x;
    var middle = (begin + end) / 2;
    while (begin <= end){
        if (x / parseInt(middle) === parseInt(middle)) {
            return parseInt(middle);
        } else if (x / parseInt(middle) > parseInt(middle)) {
            begin = parseInt(middle) + 1;
        } else {
            end = parseInt(middle) - 1;
        }
        middle = (begin + end) / 2;
    }
    return parseInt(middle);
};

console.log(sqrt(2));
console.log(sqrt(4));
console.log(sqrt(6));
console.log(sqrt(8));
console.log(sqrt(9));
console.log(sqrt(10));


